### PR TITLE
88609 Update transformer to pull patient comments from the new additionalAppointmentDetails field - VA Requests

### PR DESCRIPTION
**Pull request by vbahinwillit**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**: Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
- _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

## Summary
This PR retrieves appointment request comments from new 'additinalAppointmentDetails' field.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#88609

## Testing done
Unit & e2e

## Screenshots
```
{
    "data": {
        "id": "ba363b44785aedee5d9cd2dadf151af842df382019c2f7a1cbadce8ac3b7b9c0",
        "type": "appointments",
        "attributes": {
            "id": "ba363b44785aedee5d9cd2dadf151af842df382019c2f7a1cbadce8ac3b7b9c0",
            "identifier": [
                {
                    "system": "Appointment/",
                    "value": "523938333135333238"
                },
                {
                    "system": "http://www.va.gov/Terminology/VistADefinedTerms/409_85",
                    "value": "983:15328"
                }
            ],
            "kind": "telehealth",
            "status": "proposed",
            "serviceType": "homeSleepTesting",
            "serviceTypes": [
                {
                    "coding": [
                        {
                            "system": "http://veteran.apps.va.gov/terminologies/fhir/CodeSystem/vats-service-type",
                            "code": "homeSleepTesting"
                        }
                    ]
                }
            ],
            "serviceCategory": [
                {
                    "coding": [
                        {
                            "system": "http://www.va.gov/Terminology/VistADefinedTerms/409_1",
                            "code": "REGULAR",
                            "display": "REGULAR"
                        }
                    ],
                    "text": "REGULAR"
                }
            ],
            "reasonCode": {
                "text": "station id: 983GC|preferred modality: VIDEO|phone number: 2013004000|email: VETERAN@ABC123.COM|preferred dates:03/28/2024 PM|reason code:QUESTIONMEDS|comments:concern telehealth"
            },
            "patientIcn": "1013678363V916823",
            "locationId": "983GC",
            "start": "2024-03-28T00:00:00Z",
            "minutesDuration": 0,
            "created": "2024-03-06T18:39:53Z",
            "requestedPeriods": [
                {
                    "start": "2024-03-28T00:00:00Z",
                    "end": "2024-03-28T00:00:00Z"
                }
            ],
            "cancellable": true,
            "extension": {
                "ccLocation": {
                    "address": {}
                }
            },
            "localStartTime": "2024-03-27T18:00:00.000-06:00",
            "location": {
                "id": "983GC",
                "type": "appointments",
                "attributes": {
                    "id": "983GC",
                    "vistaSite": "983",
                    "vastParent": "983",
                    "type": "va_facilities",
                    "name": "Fort Collins VA Clinic",
                    "classification": "Multi-Specialty CBOC",
                    "timezone": {
                        "timeZoneId": "America/Denver"
                    },
                    "lat": 40.553875,
                    "long": -105.08795,
                    "website": "https://www.cheyenne.va.gov/locations/Fort_Collins_VA_CBOC.asp",
                    "phone": {
                        "main": "970-224-1550"
                    },
                    "physicalAddress": {
                        "line": [
                            "2509 Research Boulevard"
                        ],
                        "city": "Fort Collins",
                        "state": "CO",
                        "postalCode": "80526-8108"
                    },
                    "healthService": [
                        "Audiology",
                        "EmergencyCare",
                        "MentalHealthCare",
                        "PrimaryCare",
                        "SpecialtyCare"
                    ]
                }
            },
            "additionalAppointmentDetails": "concern telehealth"
        }
    }
}
```
## What areas of the site does it impact?
VAOS

## Acceptance criteria
- [ ] - cleanup logic that pulls comments.
- [ ] - tests updated

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/services/appointment/transformers.js [hunk 1]
```diff
@@ -364,12 +343,7 @@ export function transformVAOSAppointment(appt) {
       purpose.serviceName === (coding?.[0]?.code || coding) ||
       purpose.commentShort === (coding?.[0]?.code || coding),
   )?.short;
-  const comments =
-    appointmentComments.length > 0 ? appointmentComments[0] : appt.reasonCode;
-  const reasonCodeText = getReasonCodeDS(appt, 'comments')
-    ? getReasonCodeDS(appt, 'comments')
-    : comments?.text;
-  const text = appt.reasonCode ? reasonCodeText : null;
+  const text = appt.reasonCode ? appt.additionalAppointmentDetails : null;
```
Comment (simiadebowale) [inline src/applications/vaos/services/appointment/transformers.js:346]: should we rename this to patientComments or something a little more descriptive that text?

**Review by simiadebowale — COMMENTED** [review]
just left a minor comment

File: src/applications/vaos/services/appointment/transformers.js (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/services/appointment/transformers.js:346]: Not quite sure I understand. Are you talking about renaming 'appt.additonalAppointmentDetails' to 'patientComments' on the backend? That makes sense to me. Then I can change the name here.

**Review by micahchiang — DISMISSED** [review]

File: src/applications/vaos/services/appointment/transformers.js (hunk 1 above)
Comment (JunTaoLuo) [reply src/applications/vaos/services/appointment/transformers.js:331]: I think @simiadebowale meant `text` => `patientComments`

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.js:331]: yes, sorry my comment wasn't clear enough

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.js:331]: @JunTaoLuo - in vets-api maybe `patientComments` is much clearer than `additionalAppointmentDetails` as well, what do you think?

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.js [hunk 2]
```diff
@@ -337,7 +339,7 @@ describe('VAOS Page: ConfirmedAppointmentDetailsPage with VAOS service', () => {
         name: 'You shared these details about your concern',
       }),
     ).to.be.ok;
-    expect(screen.getByText(/New medical issue: I have a headache/)).to.be.ok;
+    expect(screen.getByText(/New medical issue/)).to.be.ok;
```
Comment (JunTaoLuo) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.js:342]: Just checking that we're intentionally changing the behavior being tested here? I think it does make sense to test the case where coding is present but text (i.e. what's returned by `additionalAppointmentDetails)` is missing.

File: src/applications/vaos/services/appointment/transformers.js (hunk 1 above)
Comment (JunTaoLuo) [reply src/applications/vaos/services/appointment/transformers.js:331]: Yea that makes sense to me too. Just FYI that we need to delay this PR to next sprint but IMO getting it done right is better than getting it done quickly 😄

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.js:331]: yup, I agree with that.

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.js:331]: @vbahinwillit - so we can update this PR `additionalAppointmentDetails` to `patientComments` but hold off on merging it until the vets api field is update

Comment (vbahinwillit) [reply src/applications/vaos/services/appointment/transformers.js:331]: Thanks for the review!

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.js (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.js:342]: Agreed!

**Comment by JunTaoLuo** [comment]
Changes LGTM! But I'll hold off on approving it until https://github.com/department-of-veterans-affairs/vets-api/pull/17687 is merged and deployed.

**Review by JunTaoLuo — DISMISSED** [review]
The API changes have been merged and deployed to prod now so we can proceed this with PR. Just one small comment left. Also do you mind doing validation for the rename ticket since this issue is so closely related?

File: src/applications/vaos/services/appointment/transformers.js [hunk 3]
```diff
@@ -311,12 +290,7 @@ export function transformVAOSAppointment(appt) {
       purpose.serviceName === (coding?.[0]?.code || coding) ||
       purpose.commentShort === (coding?.[0]?.code || coding),
   )?.short;
-  const comments =
-    appointmentComments.length > 0 ? appointmentComments[0] : appt.reasonCode;
-  const reasonCodeText = getReasonCodeDS(appt, 'comments')
-    ? getReasonCodeDS(appt, 'comments')
-    : comments?.text;
-  const text = appt.reasonCode ? reasonCodeText : null;
+  const text = appt.reasonCode ? appt.patientComments : null;
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/services/appointment/transformers.js]: Done

**Review by JunTaoLuo — APPROVED** [review]

**Pull request merged**